### PR TITLE
test for set_conda_env_names_and_paths

### PR DESCRIPTION
**Pull request by jemten**

### CODE REVIEW

**Comment by jemten** [comment]
The regex previously used to exclude the "t" directory also excluded directories ending on t, such as the "Set" dir. Fixed in the last commit

**Comment by jemten** [comment]
Missed a test for 100% coverage of that sub. Should be fixed now

File: t/set_conda_env_names_and_paths.t [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    conda_dir_path => catdir( $Bin, qw{ data modules miniconda } ),
+);
+
+set_conda_env_names_and_paths(
+    {
+        parameter_href => \%parameter,
+        log            => $log,
+    }
+);
+
```
Comment (henrikstranneheim) [inline t/set_conda_env_names_and_paths.t]: Could you use the given/when/then syntax to put the tests in some context

File: t/set_conda_env_names_and_paths.t [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        emip       => q{Test},
+        test_env_1 => undef,
+        test_env_2 => q{test_2},
+    },
+    conda_dir_path => catdir( $Bin, qw{ data modules miniconda } ),
+);
+
+set_conda_env_names_and_paths(
+    {
+        parameter_href => \%parameter,
```
Comment (henrikstranneheim) [inline t/set_conda_env_names_and_paths.t]: Order Ab

File: t/set_conda_env_names_and_paths.t [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+## Creates log object
+my $log = initiate_logger(
+    {
+        file_path => $test_log_path,
+        log_name  => q{TEST},
+    }
+);
+
+my %parameter = (
+    installations    => [qw{ emip test_env_1 test_env_2 }],
```
Comment (henrikstranneheim) [inline t/set_conda_env_names_and_paths.t]: Order AB

File: t/set_conda_env_names_and_paths.t [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+);
+
+## Test undef main env name
+$parameter{environment_name}{emip}       = undef;
+$parameter{environment_name}{test_env_1} = undef;
+
+trap {
+    set_conda_env_names_and_paths(
+        {
+            parameter_href => \%parameter,
```
Comment (henrikstranneheim) [inline t/set_conda_env_names_and_paths.t]: Order AB

**Comment by henrikstranneheim** [comment]
👍

**Pull request merged**